tests/ovm_test.go

```diff
@@ -0,0 +1,341 @@
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vm.OwnerReferences).To(BeEmpty())
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("shloud recreate VM if it gets deleted", func() {
+			newOVM := newOfflineVirtualMachine(true)
+			// Delete the VM
+			Eventually(func() error {
+				return virtClient.VM(newOVM.Namespace).Delete(newOVM.Name, &v12.DeleteOptions{})
+			}, 120*time.Second, 1*time.Second).Should(Succeed())
+
+			Eventually(func() bool {
+				_, err := virtClient.VM(newOVM.Namespace).Get(newOVM.Name, v12.GetOptions{})
+				if errors.IsNotFound(err) {
+					return false
+				}
+				return true
+			}, 120*time.Second, 1*time.Second).Should(BeTrue())
+		})
+
+		It("should stop VM if running set to false", func() {
+
+			currOVM := newOfflineVirtualMachine(false)
+			currOVM = startOVM(currOVM)
+			currOVM = stopOVM(currOVM)
+
+		})
+
+		It("should start and stop VM multiple times", func() {
+			var currOVM *v1.OfflineVirtualMachine
+
+			currOVM = newOfflineVirtualMachine(false)
+
+			// Start and stop VM multiple times
+			for i := 0; i < 5; i++ {
+				By(fmt.Sprintf("Doing run: %d", i))
+				startOVM(currOVM)
+				stopOVM(currOVM)
+			}
+		})
+
+		It("should not update the VM spec if Running", func() {
+			newOVM := newOfflineVirtualMachine(true)
+
+			Eventually(func() bool {
+				newOVM, err = virtClient.OfflineVirtualMachine(newOVM.Namespace).Get(newOVM.Name, &v12.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return hasCondition(newOVM, v1.OfflineVirtualMachineRunning)
+			}, 360*time.Second, 1*time.Second).Should(BeTrue())
+
+			By("Updating the OVM template spec")
+			newOVM, err = virtClient.OfflineVirtualMachine(newOVM.Namespace).Get(newOVM.Name, &v12.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedOVM := newOVM.DeepCopy()
+			updatedOVM.Spec.Template.Spec.Domain.Resources.Requests = v13.ResourceList{
+				v13.ResourceMemory: resource.MustParse("4096Ki"),
+			}
+			updatedOVM, err := virtClient.OfflineVirtualMachine(updatedOVM.Namespace).Update(updatedOVM)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Expecting the old VM spec still running")
+			vm, err := virtClient.VM(newOVM.Namespace).Get(newOVM.Name, v12.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vmMemory := vm.Spec.Domain.Resources.Requests.Memory()
+			ovmMemory := newOVM.Spec.Template.Spec.Domain.Resources.Requests.Memory()
+			Expect(vmMemory.Cmp(*ovmMemory)).To(Equal(0))
+
+			By("Restarting the OVM")
+			newOVM = stopOVM(newOVM)
+			newOVM = startOVM(newOVM)
+
+			By("Expecting updated spec running")
+			vm, err = virtClient.VM(newOVM.Namespace).Get(newOVM.Name, v12.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vmMemory = vm.Spec.Domain.Resources.Requests.Memory()
+			ovmMemory = updatedOVM.Spec.Template.Spec.Domain.Resources.Requests.Memory()
+			Expect(vmMemory.Cmp(*ovmMemory)).To(Equal(0))
+		})
+
+		It("should survive guest shutdown, multiple times", func() {
```

Comment:
Ah ok here is the test for guest-shutdown. What you should actually test is keeping the same OVM throughout the whole test livetime and making sure that it starts the VM again if the guest was shut down from inside.

Reply:
Done :)